app/src/main/java/com/sopt/now/compose/feature/auth/SignUpViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package com.sopt.now.compose.feature.auth
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.sopt.now.compose.core.view.UiState
+import com.sopt.now.compose.data.api.ApiFactory
+import com.sopt.now.compose.data.request.RequestSignUpDto
+import com.sopt.now.compose.feature.model.User
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharedFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+
+class SignUpViewModel : ViewModel() {
+
+    private val _signUpResponseState = MutableSharedFlow<UiState<User>>()
+    val signUpResponseState: SharedFlow<UiState<User>> get() = _signUpResponseState.asSharedFlow()
+
+    fun setUser(user: User) {
+        postSignUp(user)
+    }
+
+    fun postSignUp(userEntity: User) {
+        viewModelScope.launch {
```

Comment:
[P1]
CoroutineContext를 명시해주시면 좋을 것 같습니다
```suggestion
        viewModelScope.launch(Dispatcher.IO) {
```